contract-examples/test/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { ethers } from "hardhat"
+const assert = require("assert")
+
+/*
+ *
+ * The `test` function is a wrapper around Mocha's `it` function. It provides a normalized framework for running the
+ * majority of your test assertions inside of a smart-contract, using `DS-Test`.
+ * The API can be used as follows:
+ * ```js
+ * test("<test_name>", "<contract_function_name>")
+ * test("<test_name>", ["<contract_function_name>"])
+ * test("<test_name>", { method: "<contract_function_name>", overrides: {}, shouldFail: false, debug: false })
+ * test("<test_name>", [{ method: "<contract_function_name>", overrides: {}, shouldFail: false, debug: false }])
+ * test("<test_name>", [{ method: "<contract_function_name>", shouldFail: false, debug: false }], {})
+ * ```
+ * All of the above examples are equivalent.
+ * Many contract functions can be called as a part of the same test:
+ * ```js
+ * test("<test_name>", [<test_fn1>, <test_fn2>, <test_fn3>])
+ * ```
+ * Individual test functions can describe their own overrides with the `overrides` property.
+ * If an object is passed in as the third argument to `test`, it will be used as the default overrides for all test
+ * functions.
+ * The following are equivalent:
+ * ```js
+ * test("<test_name>", [{ method: "<contract_function_name>", overrides: { from: "0x123" } }])
+ * test("<test_name>", [{ method: "<contract_function_name>" }], { from: "0x123" })
+ * ```
+ * In the above cases, the `from` override must be a signer.
+ * The `shouldFail` property can be used to indicate that the test function should fail. This should be used sparingly
+ * as it is not possible to match on the failure reason.
+ * Furthermore, the `debug` property can be used to print any thrown errors when attempting to
+ * send a transaction or while waiting for the transaction to be confirmed (the transaction is the smart contract call).
+ * `debug` will also cause any parseable event logs to be printed that start with the `log_` prefix.
+ * `DSTest` contracts have several options for emitting `log_` events.
```

Comment:
These comments are basically the equivalent of what's in the `README`. Perhaps consider linking to the `README` instead?

Reply:
Fair comment. The issue is that the README is almost guaranteed to deviate from the code. Even doc-comments tend to lag actual code changes without any formal way of doing doc-tests. 

Right now, it's basically a copy-paste in the readme, and I need to add a disclaimer saying so with a link to `utils.ts`